arduino/arduino-ide
# chore: Use latest LS with the URL encoding fixes

### Motivation


The Arduino language features must be functional when there is a `#` in the main sketch file name.

### Change description


 - Update the LS version with the fix,
 - Removed an unused module from the code.

### Other information


To verify:
 - Your IDE2 installation contains a `#` in the path. The best is to use something like `C:\Users\You\Desktop\# here is your IDE` because it will ensure that after cutting the `#` (fragment) from the URI, it won't try to load VS Code extensions incorrectly from `C:\Users\You\Desktop\`.
 - You have `"arduino.language.realTimeDiagnostics": true,` in the `settings.json`,
 - Create a sketch with a `#` in the name or your sketchbook path has a `#`,
 - Select a board,
 - Make syntax errors,
 - You see the error diagnostics in the editor.
 - Create another .ino file and a .cpp lib in the sketch. The Arduino language features must be functional.


https://user-images.githubusercontent.com/1405703/211533188-32a4e479-3cef-4b1d-87da-5e2ce15a77c6.mp4


https://user-images.githubusercontent.com/1405703/212358862-4e0c784f-7f77-4b83-95b7-123f57c2a34c.mp4



Closes #1124
Closes #1815

### Reviewer checklist
